### PR TITLE
Refactor to more general library

### PR DESCRIPTION
**Pull request by ny0m**

### CODE REVIEW

File: blockcipher/block.go [hunk 1]
```diff
@@ -0,0 +1,29 @@
+// Package blockcipher keeps things simple by only allowing a 128-bit block size,
+// irrespective of key size.
+package blockcipher
```
Comment (ny0m) [inline blockcipher/block.go:3]: I'm uncertain about the name of this package, it feels a bit long. Any other ideas?

Comment (lidijarad) [reply blockcipher/block.go:3]: I think it's fine. You can shorten it to `blckcphr` if you want, but I don't think its necessary.

File: blockcipher/block.go [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+	for i := 0; i < len(bytes); i++ {
+		block[i] = bytes[i]
+	}
+
+	return block
+}
+
+// String returns a hexadecimal representation of each byte in the block.
+func (b Block) String() string {
+	return fmt.Sprintf("%02x %02x %02x %02x %02x %02x %02x %02x %02x %02x %02x %02x %02x %02x %02x %02x", b[0], b[1], b[2], b[3], b[4], b[5], b[6], b[7], b[8], b[9], b[10], b[11], b[12], b[13], b[14], b[15])
```
Comment (lidijarad) [inline blockcipher/block.go:28]: If we don't explicitly need the spaces in the string, then we can also use `hex.EncodeToString(b[:])`. Apparently it's faster than `Sprintf` due to something called reflection ([stackoverflow](https://stackoverflow.com/questions/19328957/how-to-go-from-bytes-to-get-hexadecimal)). 

Did you want the spaces for printing?

**Review by lidijarad — APPROVED** [review]
Did you want to eventually rename the repo to `crypto`?

File: blockcipher/block.go (hunk 2 above)
Comment (ny0m) [reply blockcipher/block.go:28]: Ya, the spaces just make it nicer to debug and compare two strings visually

File: blockcipher/block.go (hunk 1 above)
Comment (ny0m) [reply blockcipher/block.go:3]: Nah let's not get too 1337

**Comment by ny0m** [comment]
> Did you want to eventually rename the repo to crypto?

Yep

**Pull request merged**